src/main/java/com/wwi21sebgroup5/cinema/helper/ImageCompressor.java

```diff
@@ -1,78 +1,100 @@
 package com.wwi21sebgroup5.cinema.helper;
 
+import com.wwi21sebgroup5.cinema.exceptions.ImageCouldNotBeCompressedException;
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+
 public class ImageCompressor {
 
+
+    //Maximum size of an Image in the database
+    public static final int MAX_IMAGE_SIZE_IN_BYTE = 50000; //50kB
+
     /**
-     * Die Methoden compressImage und decompressImage können zu späterem Zeitpunkt durch funktionierende Methoden ersetzt werden
+     * @param bytes
+     * @param imageQuality Wird genutzt, um das Bild schrittweise zu komprimieren, bis die Maximalgröße unterschritten wurde, der die Qualität zu gering ist.
+     *                     Sollte beim ersten Aufruf 1 sein
+     * @param type         gibt den typ des Bilds an -> z.B.  "image/png" oder "image/jpeg"
+     * @return Byte-Array mit den komprimierten Bild Daten
+     * @throws IOException                        wenn beim Lesen und schreiben der streams etwas fehlschlägt
+     * @throws ImageCouldNotBeCompressedException wenn die Qualität des Bildes zu stark eingeschränkt wird
      */
+    public static byte[] compressImage(byte[] bytes, float imageQuality, String type) throws IOException, ImageCouldNotBeCompressedException {
```

Comment:
Hab jetzt kein Lust mich in die Thematik einzulesen, ich vertrau hier mal auf deine Recherchearbeit. :D